front/pages/w/[wId]/builder/data-sources/static.tsx

```diff
@@ -149,43 +173,67 @@ export default function DataSourcesView({
             }}
           />
         )}
-        <ContextItem.List>
-          {dataSources.map((ds) => (
-            <ContextItem
-              key={ds.name}
-              title={ds.name}
-              visual={
-                <ContextItem.Visual
-                  visual={({ className }) =>
-                    FolderOpenIcon({
-                      className: className + " text-element-600",
-                    })
-                  }
-                />
-              }
-              subElement={
-                <>
-                  Added by: {ds.editedByUser?.fullName}
-                  <span className="h-3 w-0.5 bg-element-500" />
-                  <div className="flex items-center gap-1">
-                    Used by: {dataSourcesUsage[ds.id] ?? 0}
-                    <Icon visual={RobotIcon} size="xs" />
-                  </div>
-                </>
-              }
-              onClick={() => {
-                void router.push(
-                  `/w/${owner.sId}/builder/data-sources/${ds.name}`
-                );
-              }}
-            >
-              <ContextItem.Description>
-                <div className="text-sm text-element-700">{ds.description}</div>
-              </ContextItem.Description>
-            </ContextItem>
-          ))}
-        </ContextItem.List>
+        {clickableDataSources.length > 0 && (
+          <DataTable
+            data={clickableDataSources}
+            columns={columns}
+            filter={dataSourceSearch}
+            filterColumn={"name"}
+          />
+        )}
       </Page.Vertical>
     </AppLayout>
   );
 }
+
+function getTableColumns() {
```

Comment:
Can we avoid duplicating this whole thing ?

Reply:
Kinda feel like you want to have control over the columns and this logic is really dependant on the page you're in. I agree that there are very similar but they will soon differ. (For example the logo in folders is gonna be dependant on the type of file it is and we will add the size of the object, cf [Figma](https://www.figma.com/design/QOxHwppG64GtPocpDhS6Y7/Product?node-id=8382-44389&t=ckBExOdhlqIwwxZv-0))

Reply:
But totally fine sharing this function between pages if you want.

Reply:
fair 👍